File: app/src/main/java/com/warunya/ricenearby/constant/MealTime.java
package com.warunya.ricenearby.constant;

import java.util.HashMap;

public enum MealTime {

    Breakfast("มื้อเช้า"),
    Lunch("มื้อกลางวัน"),
    Dinner("มื้อเย็น"),
    LateDinner("มื้อค่ำ");

    String mealTimeText;

    MealTime(String mealTimeText) {
        this.mealTimeText = mealTimeText;
    }

    public String getMealTimeText() {
        return mealTimeText;
    }

    public static MealTime parse(int type) {
        HashMap<Integer, MealTime> creatorMap = new HashMap<>();
        creatorMap.put(Breakfast.ordinal(), Breakfast);
        creatorMap.put(Lunch.ordinal(), Lunch);
        creatorMap.put(Dinner.ordinal(), Dinner);
        creatorMap.put(LateDinner.ordinal(), LateDinner);

        return (creatorMap.get(type) == null) ? Breakfast : creatorMap.get(type);
    }
}

File: app/src/main/java/com/warunya/ricenearby/constant/MealTime.kt
package com.warunya.ricenearby.constant

import java.util.*

enum class MealTime(val mealTimeText: String) {

    Breakfast("มื้อเช้า"),
    Lunch("มื้อกลางวัน"),
    Dinner("มื้อเย็น"),
    LateDinner("มื้อค่ำ");


    companion object {
        fun parse(type: Int): MealTime {
            val creatorMap = HashMap<Int, MealTime>()
            creatorMap[Breakfast.ordinal] = Breakfast
            creatorMap[Lunch.ordinal] = Lunch
            creatorMap[Dinner.ordinal] = Dinner
            creatorMap[LateDinner.ordinal] = LateDinner
            return if (creatorMap[type] == null) Breakfast else creatorMap[type]!!
        }
    }
}